docs/04.guides/04.cookbooks/24.Query-Handling/page.md

```diff
@@ -0,0 +1,250 @@
+<cfquery name="qry" datasource="test" params="#{title:'Susi'}#">
+	select * from Foo1890
+	where title=:title
+</cfquery>
+<cfdump var="#qry#" expand="false">
+```
+
+referenced as ```:key``` in sql.
+
+Below example to pass the more information with struct
+
+```lucee
+<cfquery name="qry" datasource="test" params="#{title:{type:'varchar',value:'Susi'}}#">
+	select * from Foo1890
+	where title=:title
+</cfquery>
+<cfdump var="#qry#" expand="false">
+```
+
+You can passed the params value by array, It referenced as `?` in SQL
+
+```lucee
+<cfquery name="qry" datasource="test" params="#['Susi']#">
+	select * from Foo1890
+	where title=?
+</cfquery>
+<cfdump var="#qry#" expand="false">
+```
+
+Below example to pass the more information with array
+
+```lucee
+<cfquery name="qry" datasource="test" params="#[{type:'varchar',value:'Susi'}]#">
+	select * from Foo1890
+	where title=?
+</cfquery>
+```
+
+## Query with script tag ##
+
+Lucee supports for two types script statements like below,
+
+```lucee
+query name="qry" datasource="test" params={title:{type:'varchar',value:'Susi'}} {
+	echo("
+		select * from Foo1890
+		where title=:title
+	");
+}
+dump(qry);
+```
+
+```lucee
+cfquery(name="qry",datasource="test",params={title:{type:'varchar',value:'Susi'}}) {
+	echo("
+		select * from Foo1890
+		where title=:title
+	");
+}
+dump(qry);
+```
+
+## QueryExecute ##
+
+It supports for both script tag & regular tag, you can pass all the arguments to the function.
+
+```lucee
+qry=queryExecute(
+	sql:"select * from Foo1890 where title=:title"
+	,options:{datasource="test"}
+	,params={title:{type:'varchar',value:'Susi'}}
+);
+dump(qry);
+```
+
+Pass the values in params same as we saw in cfquerytag, In options we can pass other arguments like datasource,cachename,dbtype
+
+
+
+### Query Component ###
+
+You can do a query with component like "Query()".
+
+```lucee
+<cfscript>
+	query=new Query(sql:"select * from Foo1890 where title=?");
+	query.setParams([{type:'varchar',value:'Susi'}]);
+	query.setDatasource('test');
+	dump(query.execute().getResult());
+</cfscript>
+```
+
+Above example we pass the sql as part with constructor. 
+Use setDatasource() function to set the datasource,
+Use setParams() function to set the param values, value used same as we see in tag
+
+```query.execute()``` function returns detail of the component, ```query.execute().getResult()``` returns query result.
+
+### Query Future ###
+We are always in discussion how to improve the functions in lucee.
+
+Here we use ```$``` instead of writeOutput or echo inside the script using like below
+
+```lucee
+<cfscript>
+query name="qry" datasource="test" params={title:{type:'varchar',value:'Susi'}} {
+	$("
+		select * from Foo1890
+		where title=:title
+	");
+}
+dump(qry);
+</cfscript>
+```
+
+We can use not only for query, you can use any where in the file ```$("Hi there")``` make our output. This idea to make a code better
+
+lucee supports for static functions, we can use like below code
```

Comment:
***toChange***
Lucee also supports static functions as shown in the example code below: